src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -38,11 +38,22 @@ void GcsResourceManager::ConsumeSyncMessage(
   // delegate the work to the main thread for thread safety.
   // Ideally, all public api in GcsResourceManager need to be put into this
   // io context for thread safety.
+
+  // We should only expect RESOURCE_VIEW messages from the syncer for GcsResourceManager.
+  // See GcsServer::InitRaySyncer for syncer registration.
+  RAY_CHECK(message->message_type() == syncer::MessageType::RESOURCE_VIEW);
+
   io_context_.dispatch(
       [this, message]() {
         rpc::ResourcesData resources;
         resources.ParseFromString(message->sync_message());
         resources.set_node_id(message->node_id());
+        if (resources.resources_total_size() == 0) {
```

Comment:
I am gracefully returning here.

Reply:
is this possible with cpu=0 case?

Reply:
we should still have other resources (e.g. memory).

Reply:
I'm not sure. But I think we can a test with this? Like print the resource here. I think it sould come with CPU/GPU and memory, i mean the basic ones at least.

Reply:
yah a node will usually come with node ip as resources: https://github.com/ray-project/ray/blob/701a6523c4121fa409f0ceab9b3e4f81f9445aeb/python/ray/_private/resource_spec.py#L160-L163